v2/fb-xcom-api-validator/src/client/SecureClient.ts

```diff
@@ -116,37 +116,44 @@ export class SecureClient {
 }
 
 export class HttpRequestWithSecurityHeaders extends BaseHttpRequest {
+  private axiosClient: AxiosInstance;
   constructor(
     private readonly securityHeadersFactory: SecurityHeadersFactory,
     config: OpenAPIConfig
   ) {
     super(config);
+    this.axiosClient = axios.create();
+    this.axiosClient.request = <T = any, R = AxiosResponse<T>, D = any>(
+      config: AxiosRequestConfig<D>
+    ): Promise<R> => {
+      const headers = this.securityHeadersFactory(config);
+      return axios.request({
+        ...config,
```

Comment:
Yes, I initially wanted to do this but it seemed redundant to me since currently static axios request is used in any case
But I agree that it makes more sense to do it that way. Will do.

Reply:
@yanivmo see https://github.com/fireblocks/fireblocks-network-link/pull/16/files#diff-8be02f63922888bcabbbcf5184c10497d404df3352e083169fd2d4ca450a1bc3R131